src/content/peerconnection/endtoend-encryption/js/videopipe.js

```diff
@@ -58,44 +41,38 @@ function VideoPipe(stream, sendTransform, receiveTransform, handler) {
         .pipeThrough(senderTransformStream)
         .pipeTo(senderStreams.writableStream);
   }
-  pc1.onicecandidate = function(event) {
-    if (event.candidate) {
-      pc2.addIceCandidate(new RTCIceCandidate(event.candidate),
-          noAction, errorHandler('AddIceCandidate'));
-    }
-  };
-  pc2.onicecandidate = function(event) {
-    if (event.candidate) {
-      pc1.addIceCandidate(new RTCIceCandidate(event.candidate),
-          noAction, errorHandler('AddIceCandidate'));
-    }
-  };
-  pc2.ontrack = function(e) {
+  this.pc2.ontrack = e => {
     if (receiveTransform) {
       const transform = new TransformStream({
         start() {},
         flush() {},
         transform: receiveTransform
       });
-      const receiverStreams = e.receiver.createEncodedVideoStreams();
+      const {receiver} = e.transceiver;
+      const receiverStreams = receiver.createEncodedVideoStreams();
```

Comment:
any particular reason why you want to go via the transceiver?

Reply:
in general because the transceiver is better for matching things (e.g. with mid) than the receiver but that isn't relevant to this example so removed it.